### DISCUSSION TITLE
Why not using the popular PiVPN Script for Wireguard?

### DISCUSSION THREAD

**Discussion by shyney7** [trinib/Adguard-Wireguard-Unbound-Cloudflare#33] [Q&A] [answered]
Hi first of all thanks for the detailed and user friendly guide for setting this all up! 
Nevertheless I would like to know if there is a specific reason not to use the PiVPN script for setting up Wireguard https://github.com/pivpn/pivpn ? I really like the idea of automated security updates.

**Comment by trinib** [comment, marked as answer]
> Hi first of all thanks for the detailed and user friendly guide for setting this all up! Nevertheless I would like to know if there is a specific reason not to use the PiVPN script for setting up Wireguard [pivpn/pivpn](https://github.com/pivpn/pivpn) ? 


PiVPN is just another script, you can use it if you like. I choose [that script](https://github.com/Nyr/wireguard-install) because I only had WireGuard in the guide at the time. I do not see any difference as long as devs keep there script updated if something in WireGuard/OpenVPN changes which is very rarely.  If you see something PiVPN has that Nyr  individual scripts don't and it's beneficial let me know.

>I really like the idea of automated security updates.

And What you mean by "automated security updates" ?

**Reply by shyney7** [reply]
I mean the unattended security updates: 

> SECURE ::: Even though this installer makes everything so trivial, it doesn't mean it gives you trivial security settings.
Everything has been upgraded right out of the box beyond the default settings to harden the security of the server and client. Starting with offering you the ability to enable unattended-upgrades which will automatically patch your server with security updates. Next, the server configuration will only use the latest TLS protocol. Both the data and control channels use upgraded AES and SHA256 encryption and hash algorithms. Options are pre-configured to verify your server certificate to battle MITM attack vectors. All this and more are configured out of the box by the pivpn installer. This is a detailed level of hardening you'll have a difficult time finding elsewhere.

https://www.pivpn.io/

**Reply by trinib** [reply]
ohh that's unattended upgrades built in the script for auto-updating PI system(via OS security patches). I have a wiki guide to do it with cronjob method and a link to manually setup unattended upgrades :  https://github.com/trinib/AdGuard-WireGuard-Unbound-Cloudflare/wiki/Auto-Update-Raspberry-Pi

It is your choice to use PiVPN if you're a longtime user .. I wanted to have everything explained manually so inexperience people can have a better understanding instead of the all in one stuff that sometimes eliminates the experience of learning the foundations.

**Reply by trinib** [reply]
@shyney7 actually I now remembered in the script I do not have ` sudo apt full-upgrade` to actually update system security patches, it can be added though. But I only have `apt-get update`/`apt-get upgrade`/`apt-get dist-upgrade` because I told myself at that time full system upgrades can cause issues from cli and I like to clean install if new version of OS and kernel comes out and reinstall everything. But its the user's choice to how they want it.
